contracts/YVaultAssetProxy.sol

```diff
@@ -252,49 +137,48 @@ contract YVaultAssetProxy is WrappedPosition {
+    function pause(bool pauseStatus) external onlyAuthorized {
+        paused = pauseStatus;
     }
 
-    /// @notice Converts an input of shares to it's output of underlying or an input
-    ///      of underlying to an output of shares, using yearn 's deposit pricing
-    /// @param amount the amount of input, shares if 'sharesIn == true' underlying if not
-    /// @param sharesIn true to convert from yearn shares to underlying, false to convert from
-    ///                 underlying to yearn shares
-    /// @dev WARNING - In yearn 0.3.1 - 0.3.5 this is an exact match for deposit logic
-    ///                but not withdraw logic in versions 0.3.2-0.3.5. In versions 0.4.0+
-    ///                it is not a match for yearn deposit ratios.
-    /// @return The converted output of either underlying or yearn shares
-    function _yearnDepositConverter(uint256 amount, bool sharesIn)
-        internal
-        view
-        virtual
-        returns (uint256)
+    /// @notice Function to transition between two yearn vaults
+    /// @param newVault The address of the new vault
+    /// @param minOutputShares The min of the new yearn vault's shares the wp will receive
+    /// @dev WARNING - This function has the capacity to steal all user funds from this
+    ///                contract and so it should be ensured that the owner is a high quorum
+    ///                governance vote through the time lock.
+    function transition(IYearnVault newVault, uint256 minOutputShares)
+        external
+        onlyOwner
     {
-        // Load the yearn total supply and assets
-        uint256 yearnTotalSupply = vault.totalSupply();
-        uint256 yearnTotalAssets = vault.totalAssets();
-        // If we are converted shares to underlying
-        if (sharesIn) {
-            // then we get the fraction of yearn shares this is and multiply by assets
-            return (yearnTotalAssets * amount) / yearnTotalSupply;
-        } else {
-            // otherwise we figure out the faction of yearn assets this is and see how
-            // many assets we get out.
-            return (yearnTotalSupply * amount) / yearnTotalAssets;
-        }
+        // Load the current vault's price per share
+        uint256 currentPricePerShare = _pricePerShare();
+        // Load the new vault's price per share
+        uint256 newPricePerShare = newVault.pricePerShare();
+        // Load the current conversion rate or set it to 1
+        uint256 newConversionRate = conversionRate == 0 ? 1e18 : conversionRate;
+        // Calculate the new conversion rate, note by multiplying by the old
+        // conversion rate here we implicitly support more than 1 upgrade
+        newConversionRate =
+            (newConversionRate * newPricePerShare) /
+            currentPricePerShare;
+        // We now withdraw from the old yearn vault using max shares
+        // Note - Vaults should be checked in the future that they still have this behavior
```

Comment:
This seems like a big assumption that should be verified before creating pools. Perhaps add it to some ops checklist?